packages/xstate-wallet/src/store/store.ts

```diff
@@ -324,6 +327,27 @@ export class XstateStore implements Store {
 
   private nonceKeyFromAddresses = (addresses: string[]): string => addresses.join('::');
 
+  async getForceMoveTransactionData(channelId: string): Promise<TransactionRequest> {
+    const channelStorage = await this.backend.getChannel(channelId);
+
+    if (!channelStorage) {
+      throw new Error('Channel not found');
+    }
+
+    const {participants} = channelStorage;
+    const myAddress = participants[channelStorage.myIndex].signingAddress;
+    const privateKey = await this.backend.getPrivateKey(myAddress);
```

Comment:
I don't think we should ever be accessing `this.backend` to perform business logic. There should just be a `public getPrivateKey(address)` method on the store that's called. Otherwise, our business logic is littered with `if (!privateKey)` checks.

Reply:
Perhaps `private`? But yeah makes sense